File: sdkwork-ai-sdk-go/api/videos.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type VideosApi struct {
    client *sdkhttp.Client
}

func NewVideosApi(client *sdkhttp.Client) *VideosApi {
    return &VideosApi{client: client}
}

// List videos
func (a *VideosApi) ListVideos(query map[string]interface{}) (sdktypes.OpenAiVideoList, error) {
    raw, err := a.client.Get(AiApiPath("/videos"), query, nil)
    if err != nil {
        var zero sdktypes.OpenAiVideoList
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiVideoList](raw)
}

// Create video
func (a *VideosApi) CreateVideo(body sdktypes.VideoGenerationRequest) (sdktypes.OpenAiVideo, error) {
    raw, err := a.client.Post(AiApiPath("/videos"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.OpenAiVideo
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiVideo](raw)
}

// Retrieve video
func (a *VideosApi) RetrieveVideo(videoId string) (sdktypes.OpenAiVideo, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/videos/%s", videoId)), nil, nil)
    if err != nil {
        var zero sdktypes.OpenAiVideo
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiVideo](raw)
}

// Delete video
func (a *VideosApi) DeleteVideo(videoId string) (sdktypes.OpenAiVideoDeleteResponse, error) {
    raw, err := a.client.Delete(AiApiPath(fmt.Sprintf("/videos/%s", videoId)), nil, nil)
    if err != nil {
        var zero sdktypes.OpenAiVideoDeleteResponse
        return zero, err
    }
    return decodeResult[sdktypes.OpenAiVideoDeleteResponse](raw)
}

File: sdkwork-ai-sdk-typescript/src/api/videos.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { OpenAiVideo, OpenAiVideoDeleteResponse, OpenAiVideoList, VideoGenerationRequest } from '../types';


export class VideosApi2 {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** List videos */
  async listVideos(params?: QueryParams): Promise<OpenAiVideoList> {
    return this.client.get<OpenAiVideoList>(aiApiPath(`/v1/videos`), params);
  }

/** Create video */
  async createVideo(body: VideoGenerationRequest): Promise<OpenAiVideo> {
    return this.client.post<OpenAiVideo>(aiApiPath(`/v1/videos`), body);
  }

/** Retrieve video */
  async retrieveVideo(video_id: string | number): Promise<OpenAiVideo> {
    return this.client.get<OpenAiVideo>(aiApiPath(`/v1/videos/${video_id}`));
  }

/** Delete video */
  async deleteVideo(video_id: string | number): Promise<OpenAiVideoDeleteResponse> {
    return this.client.delete<OpenAiVideoDeleteResponse>(aiApiPath(`/v1/videos/${video_id}`));
  }
}

export function createVideosApi2(client: HttpClient): VideosApi2 {
  return new VideosApi2(client);
}
